js/source/geojson_source.js

```diff
@@ -1,28 +1,23 @@
 'use strict';
 
-var util = require('../util/util');
 var Evented = require('../util/evented');
-var TilePyramid = require('./tile_pyramid');
-var Source = require('./source');
+var util = require('../util/util');
 var urlResolve = require('resolve-url');
 var EXTENT = require('../data/bucket').EXTENT;
 
-module.exports = GeoJSONSource;
+module.exports.create = function (id, options, dispatcher) {
+    return new GeoJSONSource(id, options, dispatcher);
+};
 
 /**
- * Create a GeoJSON data source instance given an options object
- * @class GeoJSONSource
- * @param {Object} [options]
- * @param {Object|string} options.data A GeoJSON data object or URL to it. The latter is preferable in case of large GeoJSON files.
- * @param {number} [options.maxzoom=18] Maximum zoom to preserve detail at.
- * @param {number} [options.buffer] Tile buffer on each side in pixels.
- * @param {number} [options.tolerance] Simplification tolerance (higher means simpler) in pixels.
- * @param {number} [options.cluster] If the data is a collection of point features, setting this to true clusters the points by radius into groups.
- * @param {number} [options.clusterRadius=50] Radius of each cluster when clustering points, in pixels.
- * @param {number} [options.clusterMaxZoom] Max zoom to cluster points on. Defaults to one zoom less than `maxzoom` (so that last zoom features are not clustered).
```

Comment:
Discussed this above with @jfirebaugh -- see https://github.com/mapbox/mapbox-gl-js/pull/2667#issuecomment-227453069